### ISSUE TITLE
Missing NULL check of receiver object in Xcheck:jni

### ISSUE THREAD

**Issue by sjmini** [eclipse/openj9#10476]
Java -version output
--------------------

openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.18.1, JRE 11 Linux amd64-64-Bit Compressed References 20200122_441 (JIT enabled, AOT enabled)
OpenJ9 - 51a5857
OMR - 7a1b0239a
JCL - da35e0c380 based on jdk-11.0.6+10)

Summary of problem
------------------

Get<'type'>Field JNI function expects a receiver object as a second argument.
Passing NULL value as a second argument of Get<'type'>Field JNI function leads to segmentation fault on OpenJ9.
OpenJ9 does not validate NULL value when Xcheck:jni is enabled.
It would be good if OpenJ9 validate the NULL object.

Diagnostic files
----------------

[test11.zip](https://github.com/eclipse/openj9/files/5138087/test11.zip)

```
Type=Segmentation error vmState=0x00040000
J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
Handler1=00007FCBE7B050B0 Handler2=00007FCBE73F0090 InaccessibleAddress=0000000000000000
RDI=0000000000A61A00 RSI=0000000000000000 RAX=00007FCBE7DF1880 RBX=0000000000A61A00
RCX=00000000000001E0 RDX=00007FCBE83C6C70 R8=0000000000000004 R9=0000000000A61A00
R10=00007FCBE80008D0 R11=0000000000000000 R12=0000000000000000 R13=00007FCBE800F490
R14=0000000000000000 R15=0000000000000004
RIP=00007FCBE7B0E330 GS=0000 FS=0000 RSP=00007FCBECC362A0
EFlags=0000000000010246 CS=0033 RBP=00007FCBE83C6C70 ERR=0000000000000004
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=0000000000000000
xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm1 000000000000ff00 (f: 65280.000000, d: 3.225261e-319)
xmm2 ffffffffffffffff (f: 4294967296.000000, d: -nan)
xmm3 43e0000000000000 (f: 0.000000, d: 9.223372e+18)
xmm4 3ecd000000000000 (f: 0.000000, d: 3.457069e-06)
xmm5 7fffffffffffffff (f: 4294967296.000000, d: nan)
xmm6 402f25d56a52d07f (f: 1783812224.000000, d: 1.557389e+01)
xmm7 3d7546a072d1771b (f: 1926330112.000000, d: 1.209394e-12)
xmm8 402f25d5167b3861 (f: 377174112.000000, d: 1.557389e+01)
xmm9 3fe720fefc6a5c31 (f: 4234828800.000000, d: 7.227778e-01)
xmm10 3bbcc86800000000 (f: 0.000000, d: 6.095003e-21)
xmm11 bc6259b35b04813d (f: 1527021824.000000, d: -7.958214e-18)
xmm12 3fd4c71efc656c1d (f: 4234505216.000000, d: 3.246534e-01)
xmm13 3c7d600000000000 (f: 0.000000, d: 2.547875e-17)
xmm14 3be4241b57b0f571 (f: 1471214976.000000, d: 3.412025e-20)
xmm15 402e7f9c1e980d00 (f: 513281280.000000, d: 1.524924e+01)
Module=/opt/java/openjdk/lib/compressedrefs/libj9vm29.so
Module_base_address=00007FCBE7A6C000
Target=2_90_20200122_441 (Linux 4.15.0-88-generic)
CPU=amd64 (8 logical CPUs) (0x7d1ee1000 RAM)
```

**Comment by JasonFengJ9** [comment]
Will start with this one.

**Comment by gacholio** [comment]
I suggest adding a "non-null jobject" type since this needs to be checked in a lot of places.

**Comment by JasonFengJ9** [comment]
> I suggest adding a "non-null jobject" type since this needs to be checked in a lot of places.

Sure, will look into it.

**Comment by gacholio** [comment]
This will also be needed for #10479 along with a further type check.

**Comment by JasonFengJ9** [comment]
This can be closed since #10794 has been merged (missed `closes` in the PR).
